sim/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+cmake_minimum_required(VERSION 3.17)
+project(bentobox)
+
+# Source files
+set(SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
+set(LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib")
+set(SOURCES "${SRC_DIR}/main.cpp")
+set(TESTS "${SRC_DIR}/test.cpp")
+
+# Executable
+add_executable(${PROJECT_NAME} ${SOURCES})
+add_executable("${PROJECT_NAME}_test" ${TESTS})
+target_include_directories(${PROJECT_NAME} PRIVATE ${SRC_DIR})
+set(CMAKE_CXX_STANDARD 20)
```

Comment:
This part was incorrectly set. It should be:
```cmake
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 20)
```
Here is some code to check if the program works with C++20:

```cpp
#include <concepts>

template<typename T>
concept Hashable = requires(T a) {
    { std::hash<T>{}(a) } -> std::convertible_to<std::size_t>;
};
```

Reply:
According to the [docs](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD) `CMAKE_CXX_STANDARD`  must be set before the targets are defined. 
> https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD